runtime-modules/content-directory/src/lib.rs

```diff
@@ -1033,30 +1075,42 @@ decl_module! {
             // against the type of its Property and check any additional constraints
             Self::ensure_property_values_are_valid(&entity_controller, &new_values_for_existing_properties)?;
 
-            // Get current property values of an entity,
-            // so we can update them if new values provided present in new_property_values.
+            let class_id = entity.get_class_id();
+
+            // Get current property values of an Entity
 
             let entity_property_values = entity.get_values();
 
-            // Make updated entity_property_values from current entity_property_values and new_property_values provided
+            let new_output_property_values = Self::make_output_property_values(new_property_values);
+
+            // Compute OutputPropertyValues, which respective Properties have unique flag set
+            // (skip PropertyIds, which respective property values under this Entity are default and non required)
+            let new_unique_property_values = OutputValuesForExistingProperties::from(&class_properties, &new_output_property_values)?.compute_unique();
+
+            // Ensure all provided Properties with unique flag set are unique on Class level
+            Self::ensure_properties_unique_option_satisfied(class_id, &new_unique_property_values)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Update property values, that should be unique on Class level
+            Self::add_unique_property_values(class_id, new_unique_property_values);
```

Comment:
I believe that there is cleanup processing missing that removes possible old values from the uniqueness index. The tests should be updated to catch this case, and all instances of such missing logic across the extrinsic should be considered.

Reply:
Only default non required unique  value hashes can be removed on `update_entity_property_values` operation. Implemented here https://github.com/Joystream/joystream/pull/1064/commits/883863befbbdad929dd1bb0db11714ec96c555be#diff-0a165b042aa89a04d896ad36b8a33af5R1027
Additional failure case for the previously discussed failure case https://github.com/Joystream/joystream/pull/1064/commits/fb19f628aa35f195d4bb94eb8d226129a59a1f5f